app/views/datasets/_form.html.erb

```diff
@@ -0,0 +1,34 @@
+<%= form_with(model: dataset) do |form| %>
+  <% if dataset.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(dataset.errors.count, "error") %> prohibited this dataset from being saved:</h2>
+      <ul>
+        <% dataset.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= form.label :title %><br>
+    <%= form.text_field :title %>
+  </div>
+
+  <div class="field">
+    <%= form.label :collection %><br>
+    <%= form.collection_select :collection_id, current_user.submitter_collections, :id, :title, {:include_blank => true}, {:class => 'form-control'} %>
```

Comment:
Notice that the dropdown with the possible collections that a user can assign a dataset to is filtered to whatever they have access to submit to via the `current_user.submitter_collections`

Reply:
👍🏻